### ISSUE TITLE
`let-with-type-underscore`  is removing `)` causing syntax error

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#15377]
Using the following flags

`--force-warn clippy::let-with-type-underscore`

this code:

```rust
// check-pass

fn main() {
    // Test that we can infer the type of binary operands when
    // references are involved, on various types and operators.
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;

    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): f32 = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;

    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + 0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0;
    let (f32): _ = 0.0 + &0.0 + &0.0 + &0.0;

    // Test type inference when variable types are indirectly inferred.
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;

    // When we have no expected type, the types of the operands is  let _: "f32 = 0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0;
}

```

caused the following diagnostics:

```
    Checking _f907ed1a9b68ff135ebfa7e0da493a42205e97ed v0.1.0 (/tmp/icemaker_global_tempdir.f0JXFBTcO4io/icemaker_clippyfix_tempdir.TQniHnY3bjCa/_f907ed1a9b68ff135ebfa7e0da493a42205e97ed)
warning: variable declared with type underscore
  --> src/main.rs:19:5
   |
19 |     let (f32): _ = 0.0 + &0.0 + &0.0 + &0.0;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_with_type_underscore
   = note: requested on the command line with `--force-warn clippy::let-with-type-underscore`
help: remove the explicit type `_` declaration
   |
19 -     let (f32): _ = 0.0 + &0.0 + &0.0 + &0.0;
19 +     let (f32 = 0.0 + &0.0 + &0.0 + &0.0;
   |

warning: `_f907ed1a9b68ff135ebfa7e0da493a42205e97ed` (bin "_f907ed1a9b68ff135ebfa7e0da493a42205e97ed") generated 1 warning (run `cargo clippy --fix --bin "_f907ed1a9b68ff135ebfa7e0da493a42205e97ed"` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.36s

```

However after applying these diagnostics, the resulting code:

```rust
// check-pass

fn main() {
    // Test that we can infer the type of binary operands when
    // references are involved, on various types and operators.
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;

    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): f32 = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;

    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + 0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0;
    let (f32 = 0.0 + &0.0 + &0.0 + &0.0;

    // Test type inference when variable types are indirectly inferred.
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let f32: (f32) = 0.0 + &0.0 + &0.0 + &0.0;

    // When we have no expected type, the types of the operands is  let _: "f32 = 0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0 + &0.0 + &0.0;
    let (f32): (f32) = 0.0 + &0.0;
}

```
no longer compiled:

```
    Checking _f907ed1a9b68ff135ebfa7e0da493a42205e97ed v0.1.0 (/tmp/icemaker_global_tempdir.f0JXFBTcO4io/icemaker_clippyfix_tempdir.TQniHnY3bjCa/_f907ed1a9b68ff135ebfa7e0da493a42205e97ed)
error: mismatched closing delimiter: `}`
  --> src/main.rs:19:9
   |
 3 | fn main() {
   |           - closing delimiter possibly meant for this
...
19 |     let (f32 = 0.0 + &0.0 + &0.0 + &0.0;
   |         ^ unclosed delimiter
...
29 | }
   | ^ mismatched closing delimiter

error: could not compile `_f907ed1a9b68ff135ebfa7e0da493a42205e97ed` (bin "_f907ed1a9b68ff135ebfa7e0da493a42205e97ed" test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_f907ed1a9b68ff135ebfa7e0da493a42205e97ed` (bin "_f907ed1a9b68ff135ebfa7e0da493a42205e97ed") due to 1 previous error

```

Version:
```
rustc 1.90.0-nightly (cc0a5b730 2025-07-31)
binary: rustc
commit-hash: cc0a5b73053c62a3df5f84b3ee85079c9b65fa87
commit-date: 2025-07-31
host: x86_64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.8

```
